lib/hooks/useMarketsUrlQuery.ts

```diff
@@ -18,15 +19,20 @@ export const useMarketsUrlQuery = (): MarketListQuery & {
   updateQuery: MarketListQueryUpdater;
 } => {
   const router = useRouter();
-  const rawQuery = router.query;
-
+  const routerPath = router.asPath;
   const query = useMemo(() => {
     try {
-      return parse(rawQuery);
+      const url = parseUri(routerPath);
```

Comment:
Have you checked if it's possible to do this with `next/router`?

Reply:
Yes, the issue with `router.query` is it is initially empty regardless of query is set initially or no, so this way we don't need to have skeletons on market filters nor the `debouncedQueryChange` state on MarketsList.